src/Microsoft.Health.Dicom.Core/Features/Store/StoreResponseBuilder.cs

```diff
@@ -17,16 +20,21 @@ namespace Microsoft.Health.Dicom.Core.Features.Store;
 public class StoreResponseBuilder : IStoreResponseBuilder
 {
     private readonly IUrlResolver _urlResolver;
+    private readonly bool _enableDropInvalidDicomJsonMetadata;
 
     private DicomDataset _dataset;
 
     private string _message;
 
-    public StoreResponseBuilder(IUrlResolver urlResolver)
+    public StoreResponseBuilder(
+        IUrlResolver urlResolver,
+        IOptions<FeatureConfiguration> featureConfiguration
```

Comment:
Should the response builder know about feature stuffs? Or should we pass as a parameter?

Reply:
I don't have strong opinion, but going with the name of the class. It feels like it should just build what the caller tells rather than importing configs. I see both the patterns followed in the code base.

Reply:
Agree - it makes sense to just evaluate the feature in the `StoreService`.